src/main/java/com/epam/esm/controller/UsersController.java

```diff
@@ -0,0 +1,91 @@
+package com.epam.esm.controller;
+
+import com.epam.esm.dto.OrderDto;
+import com.epam.esm.dto.UserDto;
+import com.epam.esm.entity.Order;
+import com.epam.esm.entity.Tag;
+import com.epam.esm.entity.User;
+import com.epam.esm.service.UserService;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.hateoas.CollectionModel;
+import org.springframework.hateoas.Link;
+import org.springframework.http.HttpStatus;
+import org.springframework.http.ResponseEntity;
+import org.springframework.web.bind.annotation.*;
+import java.util.List;
+
+import static org.springframework.hateoas.server.mvc.WebMvcLinkBuilder.linkTo;
+import static org.springframework.hateoas.server.mvc.WebMvcLinkBuilder.methodOn;
+
+/**
+ * The class that represents an API for basic operations with the application concerned to users.
+ *
+ * @author Aliaksei Bliznichenka
+ */
+@RestController
+@RequestMapping("/users")
+public class UsersController {
+
+    private static final String JSON = "application/json";
+    private static final String ID = "id";
+    private static final String ALL_USERS = "users";
+    private static final String ALL_ = "users";
```

Comment:
Why do we need these constants?

Reply:
ID is used as a constant in @PathVariable. Others not needed, so I removed them